canvas.go

```diff
@@ -503,54 +501,17 @@ func (c *Canvas) SaveTeX(filename string) error {
 	return tex.Close()
 }
 
-// WriteImage saves the canvas as a rasterized image with given DPM (dots-per-millimeter). Higher DPM will result in bigger images.
-func (c *Canvas) WriteImage(dpm float64) *image.RGBA {
-	img := image.NewRGBA(image.Rect(0, 0, int(c.W*dpm+0.5), int(c.H*dpm+0.5)))
-	ras := NewRasterizer(img, dpm)
-	c.Render(ras)
-	return img
-}
-
-// SavePNG saves the canvas to a PNG file.
-func (c *Canvas) SavePNG(filename string, dpm float64) error {
-	f, err := os.Create(filename)
-	if err != nil {
-		return err
-	}
-
-	img := c.WriteImage(dpm)
-	// TODO: optimization: cache img until canvas changes
-	if err = png.Encode(f, img); err != nil {
-		f.Close()
-		return err
-	}
-	return f.Close()
-}
-
-// SaveJPG saves the canvas to a JPG file.
-func (c *Canvas) SaveJPG(filename string, dpm float64, opts *jpeg.Options) error {
-	f, err := os.Create(filename)
-	if err != nil {
-		return err
-	}
-
-	img := c.WriteImage(dpm)
-	if err = jpeg.Encode(f, img, opts); err != nil {
-		f.Close()
-		return err
-	}
-	return f.Close()
-}
+// Writer can write a canvas to a writer
+type Writer func(w io.Writer, c *Canvas) error
```

Comment:
I like this abstraction!

Reply:
So do I :smiley: it somehow emerged while trying to find an elegant solution to the gochart stuff.